health/health.go

```diff
@@ -170,6 +170,20 @@ func Register(name string, check Checker) {
 	registeredChecks[name] = check
 }
 
+// Unregister removes the named checker.
+func Unregister(name string) {
```

Comment:
@aaronlehmann I had this same problem when adding the driver health check. Generally, we are following the model of global health check registry to maintain uniqueness for the process. Obviously, this is at odds with unit testing. We may want to look at the `flag` package or the `net/http` package for testing feedback on global registration patterns.

What if we change the signature of `RegisterHealthChecks` to `RegisterHealthChecks(registerFn func(name string, check Checker))`? The main file passes in `health.Register` and the unit tests pass in their own registry than can be checked for correctness.

The other option would be to export a `health.Registry` object that can be used to manage registrations. `health.Register` calls into the `DefaultRegistry` but unit tests can define a separate one. This is more work but probably would provide a better result.